### ISSUE TITLE
Add support for Delta 3 Plus units

### ISSUE THREAD

**Issue by jstewart612** [tolwi/hassio-ecoflow-cloud#395]
Add support for Delta 3 Plus units.  If there's any steps I can follow to collect data on my units, glad to help gather data for anyone willing to contribute.  I just don't know how offhand.

**Comment by nalditopr** [comment]
This will get you started:

https://gist.github.com/foxthefox/11ac35433c228085a8f5e25d5651632a?permalink_comment_id=5316309#gistcomment-5316309

**Comment by AlphaZuluNRW** [comment]
If I see this correctly, then the public API is not chosen here, but rather access via the APP, which is no longer supported by ecoflow.

Is there no way to do that via the public API? Unfortunately, there doesn't seem to be anything more coming from @tolwi on the matter, as the last activity dates back to December.

**Comment by AlphaZuluNRW** [comment]
Has anyone managed to integrate the "Delta 3" via the official EcoFlow API? Unfortunately, the above-mentioned “solution” via NodeRed doesn’t work for me. I also couldn't find a solution to the problem via Google.

**Comment by nalditopr** [comment]
I'll share an updated node red file. There's an array that must be parsed and a new protobuf file.

**Comment by nalditopr** [comment]
@AlphaZuluNRW https://gist.github.com/nalditopr/426f662d1247e898e847429e220baa6e

**Comment by nalditopr** [comment]
> If I understand correctly, it is not the public API that is chosen here, but rather access via the APP, which is no longer supported by ecoflow.
> 
> Isn't there a way to do this via the public API? Unfortunately, there doesn't seem to be anything more coming from [@tolwi](https://github.com/tolwi) on the matter, as the last activity dates back to December.

Unfortunately they have not released the official API for it just yet. Once they do, it would be very simple to ass to the current integration from @tolwi 

This is an EcoFlow issue.

**Comment by AlphaZuluNRW** [comment]
> Unfortunately they have not released the official API for it just yet.

But EcoFlow offers this for the Delta 3 pro: [https://developer-eu.ecoflow.com/us/document/deltaPro3](url)

Or is that something else?

**Comment by nalditopr** [comment]
@AlphaZuluNRW Right, that's the Pro. Took them almost 6 months to release the API after its release. This thread is about the Plus, which came to market Q4 of last year. No API for the Plus, only MQTT/APP.

**Comment by foxthefox** [comment]
The new Delta 3 series is communicating with protobuf messages, which are not documented. But the telegrams may be decoded by educated guesses knowing the working conditions. 

There is a workaround using the HA MQTT discovery feature and using an instance of ioBroker (another homeautomation system).

I have made a small picturized description.
I hope it helps

https://github.com/foxthefox/ioBroker.ecoflow-mqtt/blob/main/doc/en/IOB_HA/navi.md

You may want to contribute on evaluating the Delta 3 series, here
https://github.com/foxthefox/ioBroker.ecoflow-mqtt/issues/181

BTW.
For Delta Pro 3 the commands will be part of the next version of the adapter.

**Comment by johnculviner** [comment]
If you want something that runs only in HA and is pretty much feature complete take a look at this. https://github.com/rabits/ha-ef-ble It's working perfectly for me and supports TOU switching with this PR (this was a big thing for me to automate) https://github.com/rabits/ha-ef-ble/pull/60

**Comment by BenedictLang** [comment]
The documentation still seems to be missing. For me only cloud API with app account works with Delta 3 as closest device. Some values might stay 0 and I can not set the charging power but everything else works.

Public API doesnt work. There is no Delta 3 option and the Delta Pro 3 does not give any info.
